src/key.rs

```diff
@@ -41,8 +41,8 @@ pub trait PrivateKey: DecryptionPrimitive + PublicKeyParts {}
     serde(crate = "serde_crate")
 )]
 pub struct RSAPublicKey {
-    n: BigUint,
-    e: BigUint,
+    pub(crate) n: BigUint,
+    pub(crate) e: BigUint,
```

Comment:
For `n` and `e` there are methods that you can call, no need to expose them `key.n()` and `key.e()`

Reply:
Yeah, just seemed weird to use the trait that's associated with it for those two and expose others, I'll switch now tho

Reply:
yeah, we might want to add a matching trait for the private parts, @str4d thoughts?

Reply:
We don't necessarily have access to the private components, e.g. if the private key is on a YubiKey (#32), so we couldn't have this be part of any of the base traits.